### PR TITLE
Improve error when read_xxx functions fail

### LINKED ISSUE #106: Misleading error when using `read_xxx` functions with invalid URLs
I get ```ERROR:  Function `read_csv(TEXT)` only works with Duckdb execution.``` when using `read_csv` (along with the actual error as a warning):

```
y=# SELECT * from read_csv('https://www.data.gouv.fr/fr/datasets/r/6d186965-f41b-41f3-9b23-88241cc6890c') as (n int);
WARNING:  (DuckDB) HTTP Error: Unable to connect to URL "https://www.data.gouv.fr/fr/datasets/r/6d186965-f41b-41f3-9b23-88241cc6890c": 404 (Not Found)
ERROR:  Function `read_csv(TEXT)` only works with Duckdb execution.
CONTEXT:  PL/pgSQL function read_csv(text) line 3 at RAISE
```

I would expect the `WARNING` to be the `ERROR` and the `ERROR` to not show.

### PR DESCRIPTION
**Pull request by JelteF**
Previously the error would be shown as a WARNING and then the query would be run as a fallback using the Postgres executor, which would fail because the DuckDB `read_csv`/`read_parquet` functions cannot be executed by the Postgres executor. This is quite confusing as many people only look at the last ERROR, and not the warnings before.

This is also relevant for MotherDuck support, since the MotherDuck backed tables can only be read using the DuckDB executor.

Fixes #106
Related to #209 (the error there looks terribly confusing)

### CODE REVIEW

File: include/pgduckdb/pgduckdb_planner.hpp [hunk 1]
```diff
@@ -11,6 +11,6 @@ extern "C" {
 
 extern bool duckdb_explain_analyze;
 
-PlannedStmt *DuckdbPlanNode(Query *parse, int cursor_options);
+PlannedStmt *DuckdbPlanNode(Query *parse, int cursor_options, int elevel);
```
Comment (Y--) [inline include/pgduckdb/pgduckdb_planner.hpp:14]: nit
```suggestion
PlannedStmt *DuckdbPlanNode(Query *parse, int cursor_options, int elevel_on_error);
```

File: src/pgduckdb_hooks.cpp [hunk 2]
```diff
@@ -166,17 +166,13 @@ IsAllowedStatement(Query *query, bool throw_error = false) {
...
-				return duckdbPlan;
-			}
-		}
-
 		if (NeedsDuckdbExecution(parse)) {
 			IsAllowedStatement(parse, true);
 
-			PlannedStmt *duckdbPlan = DuckdbPlanNode(parse, cursor_options);
+			PlannedStmt *duckdbPlan = DuckdbPlanNode(parse, cursor_options, ERROR);
+			return duckdbPlan;
```
Comment (Y--) [inline src/pgduckdb_hooks.cpp:173]: nit
```suggestion
			return DuckdbPlanNode(parse, cursor_options, ERROR);
```

**Review by Y-- — APPROVED** [review]
LGTM

File: include/pgduckdb/pgduckdb_planner.hpp (hunk 1 above)
Comment (JelteF) [reply include/pgduckdb/pgduckdb_planner.hpp]: I changed this to `bool throw_error` for consistency with `IsAllowedStatement`.

**Review by Y-- — APPROVED** [review]

**Pull request merged**